numbers/logSpace.go

```diff
@@ -90,6 +90,9 @@ func LogPow(x float64, y float64) float64 {
 	if x < 0 {
 		log.Fatalf("LowPowInt does not handle negative x values. x=%e\n", x)
 	}
+	if y == 0.0 {
```

Comment:
One small improvement I can come up with is rearranging the order of the if statements to return `y*math.Log(x)` first; followed by the log.Fatal and y==0 will help exit out of the function sooner. something to start thinking about is the order of your logical equivalent if statements influence the behavior of function slightly

something like:
```
if x > 0 && y != 0.0
      return y * math.Log(x)
else
      if x < 0
             log.Fatal
      if y == 0.0
             return 0.0
```

Reply:
I am interested to learn more about this at code review.  I would have naively thought that either way the code's "path" would have been to 1) test x, 2) test y, 3) return the result from the equation.  Does the speed-up come from branch prediction at the level of the processor where it assumes it will not take the branch?  Or are you changing the tests to make it faster (e.g. == to != for y)?

Reply:
As I stare at this a bit more I think in this particular case you are correct that it more or less runs the same way. I was trying to change the tests to make it faster. I was just trying to get the function to return y * math.Log(x) as quickly as possible before it has to check the other if statements.  I think the logic is more obvious if we only have to check for one variable, but since we have to check x and y anyway i think you are correct, unless we do some testing, it ends up being very similar.